packages/lib/ProtocolUtils.ts

```diff
@@ -0,0 +1,33 @@
+export function getNoteUrl(noteId: string) {
+	return `joplin://x-callback-url/openNote?id=${noteId}`;
+}
+
+export function getFolderUrl(folderId: string) {
+	return `joplin://x-callback-url/openFolder?id=${folderId}`;
+}
+
+export function getTagUrl(tagId: string) {
+	return `joplin://x-callback-url/openTag?id=${tagId}`;
+}
+
+export type Command = 'openNote' | 'openFolder' | 'openTag';
+
+export interface UlrInfo {
+    command: Command;
+    params: any;
+}
+
+export function parseUrl(s: string): UlrInfo {
+	if (!s.startsWith('joplin://')) return null;
+	const url = new URL(s);
```

Comment:
Not sure if "URL" is defined in React Native. Perhaps you could use the "url-parse" which is already part of the lib package?

Reply:
I was going to implement the same functionality for mobile next. Until then I don't think anyone is going to call this code on mobile so while it's not safe it shouldn't be an issue either.